funcs.sh

```diff
@@ -101,11 +101,19 @@ function node_build {
 
 function package_source {
 	echo "Packaging up source code"
-	WORKING_DIR=${PWD##*/}
-	cd ../
-	mkdir -p site
-	cp -rp ${WORKING_DIR}/. site
-	rm -rf site/.git/
-	tar -czf /payload-source-"$1".tgz site
-	du -h /payload-source-"$1".tgz
+	PACK_SOURCE="/tmp/site"
+	PACK_TARGET="/tmp/payload-source-$1.tgz"
+
+	# Pre-filter source, while not changing the original build dir (host may be wanting it!)
+	cp -rp . "$PACK_SOURCE"
+	rm -rf "$PACK_SOURCE/.git"
```

Comment:
Rather than manually removing `.git` sources or other sources and manually packaging a `tar`. We could use `composer archive` functionality instead as this is all build-in to this process.

E.g.

```
composer archive --format=tar --dir=/tmp --file=payload-source-$1.tar

# produces /tmp/payload-source-$1.tar archive containing codebase (Excluding files marked for exclusion by developers).
```

This uses composer functionality to package up the release (which ignores .git by default, but also allows developers to specify more "ignore" functionality through composer).

Reply:
Yeah I tried, it produces a packagist package, that doesn't contain any of the `composer install`ed files (e.g. no vendor).

Reply:
Hmm, if we were able to give notice / fall-back, then we could inform people to configure `composer.json` correctly if using the default build image.

(E.g. Correctly using `archive: {exclude:` to ensure the correct items are packaged in archive).
E.g. https://github.com/composer/composer/issues/6620#issuecomment-324123913

But seeing as most people probably are not doing this, it would cause breakages if we just expected it out of the blue. Annoying, but not much we can do about it now.

Reply:
Yeah you not going to have all the ducks in row for this one, and also `.git` is something *we* put in there so it's also something *we* should be deleting - so users shouldn't be needing to exclude that (it's platform-specific I think). 

I feel like composer archive is meant to upload packages to packagist, and not for release purposes, tbh, so I'm not so hot on the idea anyway :)

Reply:
According to composer docs, it piggybacks on the `.gitignore`  settings to determine what should be included (so `.git` folders are excluded by default). This also means that other files/folders (e.g. `vendor/` are also excluded by default as people usually exclude these from git).

Composer doesn't define an intention or use-case for it.

But either way, would be too difficult to implement today as we don't currently use this method and would require %99 of codebases to be updated with `archive` settings. So is a mute point either way.